File: Python/2020/day2/day2.py
with open("input.txt") as file:
    part1 = 0
    part2 = 0
    for x in file:
        line = x.split()
        numbers = line[0].split('-')
        print(line)
        print(f"{numbers[0]}, {line[2].count(line[1][0])}, {numbers[1]}")
        if int(numbers[0]) <= line[2].count(line[1][0]) <= int(numbers[1]):
            part1 += 1
        counter = 0
        counter += 1 if line[2][int(numbers[0])-1] == line[1][0] else 0
        counter += 1 if line[2][int(numbers[1]) - 1] == line[1][0] else 0
        if 0 < counter < 2:
            part2 += 1
print(part1)
print(part2)

File: C#/2020/day2/day2.cs
using System;
using System.Linq;

namespace day2
{
    class Program
    {
        static void Day2(string path)
        {
            string[] lines = System.IO.File.ReadAllLines(path);
            var part1 = 0;
            var part2 = 0;
            foreach(var line in lines)
            {
                string[] elements = line.Split(' ');
                var letter = elements[1][0];
                int count = elements[2].Count(x => (x == letter));
                int[] range = elements[0].Split("-").Select(x => Int32.Parse(x)).ToArray();
                if (range[0] <= count && count <= range[1]) part1++;
                var counter = 0;
                if (elements[2][range[0] - 1] == letter) counter++;
                if (elements[2][range[1] - 1] == letter) counter++;
                if (counter == 1) part2++;
            }
            Console.WriteLine($"Part1: {part1}\nPart2: {part2}");
        }
        static void Main(string[] args)
        {
            Day2("day2.txt");
        }
    }
}
